pkg/operation/botanist/component/logging/event_logger.go

```diff
@@ -0,0 +1,364 @@
+	}
+
+	if secretsManager == nil {
+		return nil, errors.New("secret manager cannot be nil")
+	}
+
+	return &eventLogger{
+		client:           client,
+		namespace:        namespace,
+		imageEventLogger: imageEventLogger,
+		secretsManager:   secretsManager,
+	}, nil
+}
+
+func (l *eventLogger) Deploy(ctx context.Context) error {
+	var (
+		serviceAccount = l.emptyServiceAccount()
+		role           = l.emptyRole()
+		roleBinding    = l.emptyRoleBinding()
+		deployment     = l.emptyDeployment()
+	)
+
+	if err := l.deployEventLoggerRBACForShootAPI(ctx); err != nil {
+		return err
+	}
+
+	// Deploy Event Logger RBAC for the Seed API
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, l.client, serviceAccount, func() error {
+		serviceAccount.AutomountServiceAccountToken = pointer.BoolPtr(false)
+		serviceAccount.Labels = getEventLoggerLabels()
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, l.client, role, func() error {
+		role.Labels = getEventLoggerLabels()
+		role.Rules = []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"events"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+		}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, l.client, roleBinding, func() error {
+		roleBinding.Labels = getEventLoggerLabels()
+		roleBinding.RoleRef = rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "Role",
+			Name:     role.Name,
+		}
+		roleBinding.Subjects = []rbacv1.Subject{{
+			Kind:      rbacv1.ServiceAccountKind,
+			Name:      serviceAccount.Name,
+			Namespace: l.namespace,
+		}}
+		return nil
+	}); err != nil {
+		return err
+	}
+
+	genericTokenKubeconfigSecret, found := l.secretsManager.Get(v1beta1constants.SecretNameGenericTokenKubeconfig)
+	if !found {
+		return fmt.Errorf("secret %q not found", v1beta1constants.SecretNameGenericTokenKubeconfig)
+	}
+
+	// Deploy the Event Logger into the shoot control-plane
+	if _, err := controllerutils.GetAndCreateOrMergePatch(ctx, l.client, deployment, func() error {
+		deployment.Labels = getEventLoggerLabels()
+		deployment.Spec.RevisionHistoryLimit = pointer.Int32Ptr(1)
+		deployment.Spec.Replicas = pointer.Int32Ptr(1)
+		deployment.Spec.Selector = &metav1.LabelSelector{
+			MatchLabels: getEventLoggerLabels(),
+		}
+		deployment.Spec.Template.ObjectMeta.Labels = getEventLoggerLabels()
+		setEventLoggerDeploymentNetworkPolicies(deployment)
+		deployment.Spec.Template.Spec.ServiceAccountName = serviceAccount.Name
+		deployment.Spec.Template.Spec.Containers = []corev1.Container{
+			{
+				Name:            eventLoggerName,
+				Image:           l.imageEventLogger,
+				ImagePullPolicy: corev1.PullIfNotPresent,
+				Command:         l.computeCommand(),
+				VolumeMounts:    l.computeVolumeMounts(),
+			},
+		}
+		deployment.Spec.Template.Spec.Volumes = l.computeVolumes(genericTokenKubeconfigSecret.Name)
```

Comment:
You could use [InjectGenericKubeconfig](https://github.com/gardener/gardener/blob/24b667c39f4ff1c6e733ce98816e4f6127a2476f/pkg/utils/gardener/shoot.go#L350) function, instead of manually creating volumes and volume mounts.